keys.go

```diff
@@ -25,7 +29,13 @@ func (ts *TxStore) GetFreshKey(purpose KeyPurpose) *hd.ExtendedKey {
 	} else {
 		index += 1
 	}
-	childKey = ts.generateChildKey(purpose, uint32(index))
+	for {
```

Comment:
I would do something like `for i :=0; i < 100; i++` so there's some sort of bound on the looping in case the error being returned is persistent (not sure what it would but seems safer than having a potentially infinite loop)